api/HttpClient.ts

```diff
@@ -146,14 +147,45 @@ export class HttpClient implements ifm.IHttpClient {
     }
 
     request(protocol: any, options: any, objs: any, onResult: (err: any, res: http.ClientResponse, contents: string) => void): void {
+        // Set up a callback to pass off 401s to an authentication handler that can deal with it
+        let self = this;
+        var callback = function (err, res, contents) {
+            if (res !== undefined && res.statusCode === 401) {
```

Comment:
Perhaps, but at the expense of having to look up the handler any on statusCode including the 401 (and not just the 401).  I'm not sure we'd need to do that.  Thoughts?